applications/tari_console_wallet/src/ui/components/transactions_tab.rs

```diff
@@ -172,6 +172,22 @@ impl TransactionsTab {
 
         let completed_txs = app_state.get_completed_txs();
         self.completed_list_state.set_num_items(completed_txs.len());
+        if let Some(detailed_tx) = &self.detailed_transaction {
+            if self.selected_tx_list == SelectedTransactionList::CompletedTxs {
+                // if we are curently viewing the completed list, we need to update the index, to match the selected
+                // index
+                let mut found_index = None;
+                let mut counter = 0;
+                for tx in &completed_txs {
+                    if detailed_tx.tx_id == tx.tx_id {
+                        found_index = Some(counter);
+                        break;
+                    }
+                    counter += 1;
+                }
```

Comment:
There's a clippy warning here, but this could be done in much more idiomatic way with: 

```
                let found_index = completed_txs
                    .iter()
                    .find(|tx| tx.tx_id == detailed_tx.tx_id)
                    .map(|&tx| tx.clone());
```